### PR TITLE
Add queries to GET /api/reviews endpoint

### PR DESCRIPTION
**Pull request by stormytuna**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -71,6 +71,106 @@ describe("GET /api/reviews", () => {
...
+
+	test("status:200, sort_by query, responds with a sorted reviews array", () => {
+		return request(app)
+			.get("/api/reviews?sort_by=votes")
+			.expect(200)
+			.then(({ body }) => {
+				const { reviews } = body;
+				expect(reviews).toHaveLength(13);
+				reviews.reduce((pre, cur) => {
+					if (pre && cur) {
```
Comment (MRombi) [inline __tests__/app.test.js:106]: We are not doing anything with the output of reduce, and pre and cur here are two objects so we could remove the reduce 🙂

File: __tests__/app.test.js [hunk 2]
```diff
@@ -71,6 +71,106 @@ describe("GET /api/reviews", () => {
...
+						review_id: expect.any(Number),
+						category: "social deduction",
+						review_img_url: expect.any(String),
+						created_at: expect.any(String),
+						votes: expect.any(Number),
+						comment_count: expect.any(String)
+					});
+				});
+			});
+	});
```
Comment (MRombi) [inline __tests__/app.test.js:173]: We should add a test for the following:
- status 400 an invalid sort_by and order i.e. sort_by=banana
- status 200 a valid category with no reviews i.e. category=children's games (res is an empty array)
- status 404 if category does not exist i.e. category=banana

**Review by MRombi — CHANGES_REQUESTED** [review]
Good work on this ticket Jack, we are almost there. Please, add the test I mentioned and implement the solution to pass them and then resubmit 🙂

File: models/models.reviews.js [hunk 3]
```diff
@@ -1,14 +1,73 @@
...
+	"title",
+	"review_id",
+	"category",
+	"review_img_url",
+	"created_at",
+	"votes",
+	"designed",
+	"comment_count"
+];
+const validOrders = ["asc", "desc"];
```
Comment (Jodiech) [inline models/models.reviews.js:14]: This needs to be moved out of the global and into the function, as you don't want to use it anywhere else.

File: models/models.reviews.js [hunk 4]
```diff
@@ -1,14 +1,73 @@
...
-    LEFT JOIN comments ON reviews.review_id = comments.review_id
-    GROUP BY reviews.review_id
-    ORDER BY created_at DESC;
-  `;
-	return db.query(query).then(({ rows: reviews }) => {
+      FROM reviews
+      LEFT JOIN comments ON reviews.review_id = comments.review_id
+    `;
+
+	// Conditionally adds filter
```
Comment (Jodiech) [inline models/models.reviews.js:37]: Please remove any comments :)

**Pull request merged**